src/dds_data.cpp

```diff
@@ -213,45 +213,56 @@ QVariant CommonData::readDynamicMember(const QString& topicName,
                                        const QString& memberName,
                                        unsigned int index)
 {
+    const char* error = "NULL";
     QVariant value;
     QMutexLocker locker(&m_dynamicSamplesMutex);
 
     if (!m_dynamicSamples.contains(topicName)) {
-        value = "NULL";
-        return value;
+        return error;
     }
 
     const QList<DDS::DynamicData_var>& sampleList = m_dynamicSamples[topicName];
-    if ((int)index >= sampleList.count()) {
-        value = "NULL";
-        return value;
+    if (static_cast<int>(index) >= sampleList.count()) {
+        return error;
     }
 
     DDS::DynamicData_var sample = sampleList.at(index);
     DDS::DynamicType_var topic_type = sample->type();
+    OpenDDS::XTypes::MemberPath member_path;
+    if (member_path.resolve_string_path(topic_type, memberName.toStdString().c_str()) != DDS::RETCODE_OK) {
```

Comment:
```suggestion
    if (member_path.resolve_string_path(topic_type, memberName.toStdString()) != DDS::RETCODE_OK) {
```

Reply:
This assumes we're not using `OPENDDS_ALLOCATOR`. I think that assumption is safe for the monitor, is that correct?

Reply:
Yes, I don't think the monitor could (or should) be built using a safety profile build of OpenDDS.

Reply:
Yeah, that's right. The safety profile build doesn't have dynamic data.